rust/README.md

```diff
@@ -0,0 +1,161 @@
+
+### Region in Cargo.toml
+Alternatively you can also specify the region in the Cargo.toml:
+
+```toml
+[workspace.metadata.cot]
+region = "NA"
+```
+
+If you do this, not specifying a region with `cargo cot build` or `cargo cot burn` will read the region from
+the Cargo.toml.
+
+## Documentation
+Most of the logic of the Rust subsystem is in the `eos-rs` crate. The documentation for this crate is available
+at <https://eosrs.pmdcollab.org/>.
+
+Broadly speaking, this crate provides two APIs:
+
+- The "high-level" API: This is a "handwritten" Rust-idiomatic way of interacting with the game's functions and data.
+  It is generally recommended to use this API.
+- The "low-level API": Everything in the `eos_rs::ffi` crate. This will give you raw access to functions and symbols
+  of the game.
+
+## Stability
+This entire subsystem is highly unstable, every update may entirely change the structure of the code. This is due
+to the fact that we are still very much in the process of reverse-engineering the game's code.
+
+Also note that a lot of code in the subsystem isn't actually fully tested yet. If you run into bugs, please report
+an issue. If it's simple to fix, consider fixing it and opening a pull request.
+
+## Usage
+To get started, read the instructions and explanations in `src/main.rs`. This file will also contain a `patches!` macro
+invocation. 
+
+One thing to note about this is that the last entry of this macro contains the string:
+```
+HasLowHealth+0:
+  B has_high_health
+```
+
+This text will be written into the file `../patches/generated_by_rust.cotpatch` when you run `cargo cot burn` and 
+will be burned into the game along with other `.cotpatch` files in that directory. These patch files contain the glue code that
+makes existing code in the game jump to your patches.
+
+This glue code is not neccesary for item or move effects or special processes as there is glue code for those
+automatically generated, see the `main.rs` file and the documentation of the 
+[`patches!` macro](https://eosrs.pmdcollab.org/armv5te-none-ndseoseabi-na/doc/eos_rs/macro.patches.html) for more 
+information.
+
+### Interop with C patches
+This runtime will compile both the Rust code in  `./src` and the C code in `../src` into the game. They are linked
+together and as such you can call Rust functions from C and vice-versa.
+
+### Logging and assertions
+You can use the normal Rust `assert!` etc. macros for assertions and for logging `eos_rs::prelude` exports the
+common `info!`, `warn!` and `error!` macros of the `log` crate. Those are written to the game's debug log, which you
+can, for example, view in the SkyTemple Script Engine Debugger.
+
+### Rust and Cargo ecosystem
+You have access to all the usual Rust / Cargo features. We encourage you to share code for patches as standalone crates,
+that other people can use!
+
+Not that the entire subsystem does not have `std` available.
+
+### Removing example code
```

Comment:
Some of the directions here pertain to the main C system, not the Rust one. Should this at least partially go in the top-level README instead? It seems weird talk about the C stuff here.

Reply:
I think it makes sense to have it here too so that people who only want to use Rust know how to clean the C code too without referring to the other README, but a link to the relevant part of the root README would also be fine imo